### PR TITLE
Devx improvements

### PR DESCRIPTION
**Pull request by GioLogist**
## Improved DevX with Yarn Workspaces

- Setup yarn workspaces, so that `example` package can import `relay-nextjs` directly 
- Update dependencies in `relay-nextjs` (dev only)
- Update typescript with live reloading, for faster devx 
- Update docs with "contributing" instructions for developing locally to contribute

---

## Contributing

If you'd like to contribute to this package, developing locally is now as simple as:

1. Run `yarn` from the root of this package
2. Start the live reloader for the package `cd packages/relay-nextjs && yarn build-w`
3. In a separate terminal strt the example app: `cd example && yarn dev`

Now, when you make any changes in `packages/relay-nextjs`, they will instantly be reflected in the example app.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -287,3 +287,13 @@ export default withRelay(UserProfile, UserProfileQuery, {
...
+
+## Contributing
+
+If you'd like to contribute to this package, developing locally is simple.
+
+1. Run `yarn` from the root of this package
+2. Start the live reloader for the package `cd packages/relay-nextjs && yarn build-w`
+3. In a separate terminal strt the example app: `cd example && yarn dev`
+
+Now, when you make any changes in `packages/relay-nextjs`, they will instantly be reflected in the example app.
```
Comment (rrdelaney) [inline README.md:299]: This is an awesome change! Thank you very much for the improvements here, it's not something I've spent a ton of time on as this repo is synced in and out of the Revere monorepo whenever changes are made. Can we add a note of which yarn version is required?

Comment (GioLogist) [reply README.md]: > This is an awesome change! Thank you very much for the improvements here, it's not something I've spent a ton of time on as this repo is synced in and out of the Revere monorepo whenever changes are made. 

No worries, happy to help!

> Can we add a note of which yarn version is required?

Good call! Done.

Comment (GioLogist) [reply README.md]: @rrdelaney Anything else needed here btw?

Comment (rrdelaney) [reply README.md]: Not at all! This PR is great, I just need to check it out locally and give it a whirl. Unfortunately I've been quite sick the past week so getting some time for that has been difficult. I will look soon though, no action need on your end, and thank you again for this wonderful effort.

Comment (GioLogist) [reply README.md]: Right on, hope you feel better soon 🤙🏻

**Pull request merged**